plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -93,10 +93,12 @@ type instance RuleResult GetHlintDiagnostics = ()
 -- |    - `getIdeas` -> `getFileContents` if the hls ghc does not match the hlint default ghc
 -- | - The client settings have changed, to honour the `hlintOn` setting, via `getClientConfigAction`
 -- | - The hlint specific settings have changed, via `getHlintSettingsRule`
-rules :: Rules ()
-rules = do
+rules :: PluginId -> Rules ()
+rules plugin = do
   define $ \GetHlintDiagnostics file -> do
-    hlintOn' <- hlintOn <$> getClientConfigAction
+    config <- getClientConfigAction
+    let pluginConfig = configForPlugin config plugin
+    let hlintOn' = hlintOn config && pluginEnabled pluginConfig plcDiagnosticsOn
```

Comment:
@jneira Did you notice that we are now *and* ing the existing `hlintOn` flag and the new one too.  At some point we should consider deprecating the old one.

Reply:
yeah, I saw it, will do a follow up to use it in the plugin